spec/requests/courses_spec.rb

```diff
@@ -1,19 +1,73 @@
 require 'helpers/rails_helper'
+require 'helpers/api_helper'
 
 RSpec.describe "Courses", type: :request do
   include_context "with authenticated requests"
-  let(:user) { create(:user) }
+  let(:student) { create(:user) }
+  let(:teacher) { create(:user, :teacher) }
   let!(:course) { create(:course) }
 
-  before do
```

Comment:
I'm trying to stay away from using variables called `user`. I think it should be very clear when a var is a student or a teacher. I dont think we can authenticate both a student and a teacher at the same time because the mock must return one value or the other.